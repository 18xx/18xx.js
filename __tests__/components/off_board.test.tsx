import { List } from 'immutable';
import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import OffBoard from '../../src/components/off_board';

describe('OffBoard', () => {
  describe('#toString', () => {
    describe('when the east exit is set', () => {
      it('draws an arrow on the right of the hex', () => {
        const subject: ReactElement<OffBoard> = (
          <OffBoard key='off-board' exits={List<number>([0])} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when the west exit is set', () => {
      it('draws an arrow on the left of the hex', () => {
        const subject: ReactElement<OffBoard> = (
          <OffBoard key='off-board' exits={List<number>([3])} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when the north west exit is set', () => {
      it('draws an arrow on the top left of the hex', () => {
        const subject: ReactElement<OffBoard> = (
          <OffBoard key='off-board' exits={List<number>([2])} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when the north east exit is set', () => {
      it('draws an arrow on the top right of the hex', () => {
        const subject: ReactElement<OffBoard> = (
          <OffBoard key='off-board' exits={List<number>([1])} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when the south west exit is set', () => {
      it('draws an arrow on the bottom left of the hex', () => {
        const subject: ReactElement<OffBoard> = (
          <OffBoard key='off-board' exits={List<number>([4])} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when the south east exit is set', () => {
      it('draws an arrow on the bottom right of the hex', () => {
        const subject: ReactElement<OffBoard> = (
          <OffBoard key='off-board' exits={List<number>([5])} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });
  });
});
