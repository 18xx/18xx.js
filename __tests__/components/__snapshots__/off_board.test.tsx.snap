// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffBoard #toString when the east exit is set draws an arrow on the right of the hex 1`] = `
<g>
  <polygon
    fill="black"
    points="110.85125168440814,60 110.85125168440814,68 96.99484522385713,64"
  />
</g>
`;

exports[`OffBoard #toString when the north east exit is set draws an arrow on the top right of the hex 1`] = `
<g>
  <polygon
    fill="black"
    points="79.67433714816835,14 86.60254037844386,18 76.2102355330306,28"
  />
</g>
`;

exports[`OffBoard #toString when the north west exit is set draws an arrow on the top left of the hex 1`] = `
<g>
  <polygon
    fill="black"
    points="24.24871130596428,18 31.17691453623979,14 34.64101615137754,28"
  />
</g>
`;

exports[`OffBoard #toString when the south east exit is set draws an arrow on the bottom right of the hex 1`] = `
<g>
  <polygon
    fill="black"
    points="86.60254037844386,110 79.67433714816835,114 76.2102355330306,100"
  />
</g>
`;

exports[`OffBoard #toString when the south west exit is set draws an arrow on the bottom left of the hex 1`] = `
<g>
  <polygon
    fill="black"
    points="31.17691453623979,114 24.24871130596428,110 34.64101615137754,100"
  />
</g>
`;

exports[`OffBoard #toString when the west exit is set draws an arrow on the left of the hex 1`] = `
<g>
  <polygon
    fill="black"
    points="0,68 0,60 13.856406460551018,64"
  />
</g>
`;
